src/lib/components/table/table-data-source.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { TableDataSource } from './table-data-source';
+import { RequestParameterInterface } from './request-parameter.interface';
+import { TerraPagerInterface } from '../pager/data/terra-pager.interface';
+import {
+    Observable,
+    of
+} from 'rxjs';
+import { TerraFilter } from './filter';
+
+class TestDataSource extends TableDataSource<{}>
+{
+    public request(requestParams:RequestParameterInterface):Observable<Array<{}>> | Observable<TerraPagerInterface<{}>>
+    {
+        return of(new Array({}));
+    }
+}
+
+interface FilterParams
+{
+    id:number;
+}
+
+describe('TableDataSource', () =>
+{
+    let dataSource:TestDataSource;
+    let filter:TerraFilter<FilterParams>;
+
+    beforeEach(() =>
+    {
+        dataSource = new TestDataSource();
+        filter = new TerraFilter<FilterParams>();
+    });
+
+    it('should create', () =>
+    {
+        expect(dataSource).toBeTruthy();
+    });
+
+    it('should create an observable for a request', () =>
+    {
+        expect(dataSource.request({})).toBeTruthy();
+    });
+
+    it('should have a filter', () =>
+    {
+        filter.filterParameter = {id: 1};
+        dataSource.filter = filter;
+
+        expect(dataSource.filter).toEqual(filter);
+    });
+
+    it('should have data after search', () =>
+    {
+        dataSource.connect(undefined).subscribe((result:Array<{}>) => expect(result).toBe([]));
+    });
```

Comment:
Mir fehlen ein paar Testfälle:
- Die property `data` wird nach einem request richtig aktualisiert
- Eine Suche wird ausgeführt, wenn
   - die `search()` Methode aufgerufen wird
   - wenn `filter.search$` emitted.
- Nachdem disconnect() aufgerufen wurde, ist die Subscription tot. D.h. wenn `search()` aufgerufen wird, wird weder die request() methode aufgerufen, noch die `data` property aktualisiert.
- Es werden die korrekten Parameter and die `request()` methode übergeben. In diese Fall wären das nur der Inhalt der `filterParameter`.

Reply:
Habe noch ein paar Tests hinzugefügt.

Reply:
Ein Test zu Punkt 4 fehlt noch.